e2e/push/push.go

```diff
@@ -75,26 +76,23 @@ func (c *ctx) testPushCmd(t *testing.T) {
 			}
 			defer os.RemoveAll(tmpdir)
 
-			cmd, out, err := e2e.ImagePush(t, c.env.CmdPath, tt.imagePath, tt.dstURI)
-			switch {
-			case tt.expectSuccess && err == nil:
-				// PASS: expecting success, succeeded
-
-			case !tt.expectSuccess && err != nil:
-				// PASS: expecting failure, failed
-
-			case tt.expectSuccess && err != nil:
-				// FAIL: expecting success, failed
-
-				t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-				t.Errorf("unexpected failure: %v", err)
-
-			case !tt.expectSuccess && err == nil:
-				// FAIL: expecting failure, succeeded
-
-				t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-				t.Errorf("unexpected success: command should have failed")
+			// We create the list of arguments using a string instead of a slice of
+			// strings because using slices of strings most of the type ends up adding
+			// an empty elements to the list when passing it to the command, which
+			// will create a failure.
+			args := tt.dstURI
+			if tt.imagePath != "" {
+				args = tt.imagePath + " " + args
 			}
+
+			e2e.RunSingularity(
```

Comment:
if you change this to `c.env.RunSingularity`, what I wrote below actually works :-|

```suggestion
			c.env.RunSingularity(
```